core/src/main/java/com/linecorp/armeria/client/retry/AbstractBackoffBuilder.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 LY Corporation
+ *
+ * LY Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.retry;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Supplier;
+
+/**
+ * A skeletal builder implementation for {@link Backoff}.
+ */
+public abstract class AbstractBackoffBuilder<SELF extends AbstractBackoffBuilder<SELF>> {
```

Comment:
Question) Does the abstract builder need to be public?

```suggestion
abstract class AbstractBackoffBuilder<SELF extends AbstractBackoffBuilder<SELF>> {
```